controllers/cluster_controller.go

```diff
@@ -357,6 +385,102 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 	return controller.Result{}, nil
 }
 
+func (r *ClusterReconciler) updateStatus(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	// An EKS-A Cluster managed by the CLI is deleted after the cluster finalizer is removed, leaving the CAPI cluster behind. In this case, we do not want to update
+	// the status. If we do, patching will through a 404 NotFound error when trying to update the status of the already deleted
+	// Cluster object.
+	if !cluster.DeletionTimestamp.IsZero() && metav1.HasAnnotation(cluster.ObjectMeta, anywherev1.ManagedByCLIAnnotation) {
+		log.Info("Clusters is managed by CLI has been deleted but CAPI cluster may remain, skipping updating cluster status")
+		return nil
+	}
+
+	// ObservedGeneration represents the last observed generation, and consequently represents that the current status is up to date.
+	// Then if observedGeneration is equal to generation AND the Cluster's "Ready" condition is "True", then we can skip another status update
+	if cluster.Status.ObservedGeneration == cluster.Generation && conditions.IsTrue(cluster, anywherev1.ReadyCondition) {
+		log.Info("Generation matches observedGeneration and the cluster is ready, skipping status update")
+		return nil
+	}
+
+	log.Info("Updating cluster status")
+	defer func() {
+		// Always update the readyCondition by summarizing the state of other conditions.
+		conditions.SetSummary(cluster,
+			conditions.WithConditions(
+				anywherev1.ControlPlaneReadyCondition,
+				anywherev1.WorkersReadyConditon,
```

Comment:
No, we want the Ready condition to be a summary of only `ControlPlaneReady` and `WorkersReady` for now. `DefaultCNIConfigured` can be false if configured to skip upgrades, so that should not be an ultimate determinator of cluster readiness. We could add `ControlPlaneInitialized`, though it would be redundant because `ControlPlaneReady` already covers for it.

Reply:
Could be redundant but maybe it's better to just include anyways if this is a summary?